bindings/wasm/src/did/wasm_document.rs

```diff
@@ -139,20 +238,84 @@ impl WasmDocument {
   ///
   /// Throws an error if the method is not found.
   #[wasm_bindgen(js_name = resolveMethod)]
-  pub fn resolve_method(&self, query: &UDIDUrlQuery) -> Result<WasmVerificationMethod> {
+  pub fn resolve_method(&self, query: &UDIDUrlQuery, scope: Option<WasmMethodScope>) -> Result<WasmVerificationMethod> {
```

Comment:
Aside: we should probably replace `resolve_method` and `resolve_method_with_scope` with a single `resolve_method` with an optional parameter...

Reply:
what do you mean? there's no `resolve_method_with_scope` here.

Reply:
Yes, I meant we should do the same thing in the Rust library proper and only have a single method.

Reply:
If having an `Option` as parameter common in Rust then that could be fine, otherwise keeping 2 methods isn't that bad IMO.

Reply:
It's actually way more than 2 methods since we have fallible and mutable variants (mostly in `CoreDocument`):

- `resolve_method`
- `try_resolve_method`
- `resolve_method_mut`
- `try_resolve_method_mut`
- `resolve_method_with_scope`
- `try_resolve_method_with_scope`

I say we replace all of them with just two methods that take in an optional `MethodScope` and return `Option` instead of `Result`:

- `resolve_method`
- `resolve_method_mut`

Reply:
Opened issue #686 for this.